In [42]:
import numpy as np
import pandas as pd
from datetime import datetime
import matplotlib.pyplot as plt

In [106]:
gdpdf = pd.read_csv('https://raw.githubusercontent.com/DanielleMaas/Incomeinequality/main/gdp.csv', usecols=['Entity', 'Year', 'GDP per capita'])
ginidf = pd.read_csv('https://raw.githubusercontent.com/DanielleMaas/Incomeinequality/main/gini_income_inequality.csv', encoding = 'latin-1', parse_dates=['year']).dropna()

In [107]:
gdpdf = gdpdf.dropna()
gdpdf['Year'] = gdpdf['Year'].astype(int)
startingyear = 1967
#startingyear is the frist year in the ginidf

gdpdf = gdpdf.loc[gdpdf['Year'] >= startingyear]
gdpdf['Year'] = pd.to_datetime(gdpdf['Year'], format='%Y')

<ipython-input-107-db12355b7532>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gdpdf['Year'] = gdpdf['Year'].astype(int)


In [108]:
gdpdf.columns= ['country', 'year', 'gdp per capita']

In [109]:
ginidf = ginidf.loc[:, ["country", "year", "gini"]]
merged = pd.merge(gdpdf, ginidf, on=["country", "year"])

In [111]:
merged.corr()

gdp per capita     gini
gdp per capita         1.00000 -0.40222
gini                  -0.40222  1.00000

Resultaat met .corr(): 
De correlatie tussen het GDP en het Gini coëfficient is -0.40222. Het is relatieve sterke maar negatieve correlatie. Een land met een lage GDP zal vaak een hoge Gini coëfficient hebben. Hieronder is nog een keer hetzelfde berekend met np.corrcoef()

In [113]:
gdp = merged.loc[: , "gdp per capita"]
gini = merged.loc[: , 'gini']
np.corrcoef(gdp, gini)

array([[ 1.        , -0.40222046],
       [-0.40222046,  1.        ]])

In [128]:
lastyear = merged.loc[merged['year'] >= "2018-01-01"]
lastyear.corr()

gdp per capita      gini
gdp per capita        1.000000 -0.413803
gini                 -0.413803  1.000000

voor het laatste jaar in het dataframe (2018) is ook nog de correlatie berekend. Dit is bijna hetzelfde, het komt uit op -0,413803. 